src/main/java/com/example/librarymanagementsystem/model/Librarian.java

```diff
@@ -0,0 +1,76 @@
+package com.example.librarymanagementsystem.model;
+
+public class Librarian {
+
+    private final int userId;
+    private final String username;
+    private final String email;
+    private final String phoneNumber;
+    private final String password;
+
+   public static class PatronBuilder{
+       private int userId;
+       private String username;
+       private String email;
+       private String phoneNumber;
+       private String password;
+
+       public PatronBuilder userId(int userId){
+           this.userId = userId;
+           return this;
+       }
+
+       public PatronBuilder username(String username){
+           this.username = username;
+           return this;
+       }
+
+       public PatronBuilder email(String email){
+           this.email = email;
+           return this;
+       }
+
+       public PatronBuilder phoneNumber(String phoneNumber){
+           this.phoneNumber = phoneNumber;
+           return this;
+       }
+
+       public PatronBuilder password(String password){
+           this.password = password;
+           return this;
+       }
+
+       public Librarian build(){
+           return new Librarian(this);
+       }
+   }
```

Comment:
Duplicate code i think. You already have a static class in the Patron class right, then you can call that builder in here instead.

eg.
`Librarian(Patron.PatronBuilder builder)`